File: server/dto/turn_context_dto.py
from dataclasses import dataclass

from dto.base_dto import BaseDto
from dto.cell_dto import CellDto
from dto.player_resources_bundle_dto import PlayerResourceBundleDto


@dataclass
class TurnContextDto(BaseDto):
    currentPlayerId: str
    isPlayer1Turn: bool
    remainingTimeInS: int
    durationInS: int
    notifyTurnChange: bool
    updatedBoardArray: list[list[CellDto]]
    playerResourceBundle: PlayerResourceBundleDto

File: client/src/dto/TurnContextDto.ts
import { EMPTY_STRING } from "../env";
import { CellDto } from "./CellDto";
import { PlayerResourceBundleDto, undefinedPlayerResourceBundleDto } from "./PlayerInfoBundleDto";

export interface TurnContextDto {
    currentPlayerId: string;
    isPlayer1Turn: boolean;
    remainingTimeInS: number;
    durationInS: number;
    notifyTurnChange: boolean;
    updatedBoardArray: CellDto[][];
    playerResourceBundle: PlayerResourceBundleDto;
}

export const undefinedTurnInfo: TurnContextDto = {
    currentPlayerId: EMPTY_STRING,
    isPlayer1Turn: false,
    remainingTimeInS: 0,
    durationInS: 0,
    notifyTurnChange: false,
    updatedBoardArray: [],
    playerResourceBundle: undefinedPlayerResourceBundleDto,
};
